src/api/users.ts

```diff
@@ -1,28 +1,51 @@
 import express from "express";
 import { authUser } from "../auth.js";
 import sql from "../database.js";
+import createHttpError from "http-errors";
+import { z } from "zod";
+import { Courses } from "../schema.js";
 
 const router = express.Router();
 export default router;
 
-router.get("/users/history", (req, res) => {
-  res.send({
-    completedCourses: [12, 52, 236],
-  });
+router.get("/users/history", async (req, res) => {
+  const user = await authUser(req);
+
+  const userCourses = await sql<Pick<Courses, "id" | "prefix" | "number">[]>`
+    SELECT course_id AS id, prefix, number FROM user_course_history
```

Comment:
Nitpick, why are we doing the `AS` here?

Reply:
Makes it weird, because we're using the `id` field from the `Course` type, but it should be `course_id`

Reply:
No, the courses table's field is id.

Reply:
Ah I see, because of the join, could be made more clear by doing `courses.id` instead of `course_id AS id`, but it's fine as is.

Reply:
Yeah, I did course_id AS id to be more explicit about getting the course_id from user_course_history.

Reply:
We have the option of using an intersection type here too, probably overkill for this situation since they have the same type, but may come in handy later on when we do joins.

`(Pick<Courses, "prefix" | "number"> & Pick<UserCourseHistory, "course_id">)[]`

Reply:
Definitely going to be useful when we do joins!